### PR TITLE
perf: cache NumberFormat before expanding options

### PR DESCRIPTION
**Pull request by ChALkeR**
Options expansion was slow

This gives additional 2.7x speedup compared to #8 

Also, we don't need lodash

### CODE REVIEW

**Review by alexandrius — APPROVED** [review]
utACK

**Review by headfire94 — APPROVED** [review]
utACK i checked we clone options before pass here

**Review by ezenwankwogabriel — APPROVED** [review]
utACK

**Comment by mvayngrib** [comment]
tests are failing for me
```
> format-num@1.1.0 unit
> ava --verbose


  ✔ edge cases
  ✖ format numbers 
  ✔ call Intl.NumberFormat correctly

  1 test failed [11:08:05]

  format numbers

  /Users/mv/code/format-num/test/test.js:22

   21:   // maybe a bug?                                        
   22:   t.is(formatNumber(10000000.00000004), '10,000,000', '')
   23:                                                          

  Difference:

  - '10,000,000.00000004'
  + '10,000,000'
```

**Comment by ChALkeR** [comment]
@mvayngrib they are failing on master (since the start)
also that test has a comment
not sure what it does and why is it there

**Review by headfire94 — APPROVED** [review]

**Comment by mvayngrib** [comment]
confirmed it also fails on master 👍

File: index.js [hunk 1]
```diff
@@ -18,22 +17,23 @@ const defaultOptions = {
...
+    if (opts && opts.nanZero === false) return 'NaN' // default is true, so we can do this without expanding the options
     else number = 0
   }
 
-  const nf = getNumberFormatter(Object.assign(Object.create(null), opts, { style: 'decimal' }))
-  return nf.format(number)
+  const key = JSON.stringify(opts)
+  if (!formatters.has(key)) {
+    opts = renameKeyShortcuts(Object.assign(Object.create(null), defaultOptions, opts))
+    opts = Object.assign(Object.create(null), opts, { style: 'decimal' })
```
Comment (mvayngrib) [inline index.js:32]: why not just `opts.style = 'decimal'` 🤔?

**Review by mvayngrib — APPROVED** [review]

File: index.js (hunk 1 above)
Comment (ChALkeR) [reply index.js:32]: it was that way, decided not to change things too much
this also removes the prototype of the opts object so there is a difference technically
not spread to not change compat
